### PR TITLE
New Account Classes and Fieldgroups for B2B CDP

### PR DESCRIPTION
**Pull request by chrispanella**
Added new activity classes for Account and their associated fieldgroups

### CODE REVIEW

File: components/classes/b2b/account-activity.schema.json [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+  },
+  "auditable": true,
+  "meta:extensible": true,
+  "meta:abstract": true,
+  "meta:extends": ["https://ns.adobe.com/xdm/data/time-series"],
+  "description": "XDM Business Account Activity is a standard Experience Data Model (XDM) class that tracks the activities related to a business account. This XDM class can only be included in the profile for customers with the B2B or B2P Edition.",
+  "definitions": {
+    "account-activity": {
+      "properties": {
+        "xdm:accountEventKey": {
```
Comment (gauravsummer) [inline components/classes/b2b/account-activity.schema.json]: Let's remove  this field.
"xdm:accountEventKey": {
So, we will only have:
"xdm:accountKey"
"xdm:accountEventType"

File: components/classes/b2b/account-history.schema.json [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        }
+      }
+    }
+  },
+  "allOf": [
+    {
+      "$ref": "https://ns.adobe.com/xdm/common/extensible#/definitions/@context"
+    },
+    {
+      "$ref": "https://ns.adobe.com/xdm/data/record"
```
Comment (gauravsummer) [inline components/classes/b2b/account-history.schema.json]: This should be similar to above class. Both are same type of class, so should include timeseries along with "meta:refProperty"

File: components/classes/b2b/account-history.schema.json [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+  },
+  "auditable": true,
+  "meta:extensible": true,
+  "meta:abstract": true,
+  "meta:extends": ["https://ns.adobe.com/xdm/data/time-series"],
+  "description": "XDM Business Account History is a standard Experience Data Model (XDM) class that tracks the change history for a business account. This XDM class can only be included in the profile for customers with the B2B or B2P Edition.",
+  "definitions": {
+    "account-history": {
+      "properties": {
+        "xdm:accountEventKey": {
```
Comment (gauravsummer) [inline components/classes/b2b/account-history.schema.json]: Let's remove this field.
"xdm:accountEventKey": {
So, we will only have:
"xdm:accountKey"
"xdm:accountEventType"

File: components/classes/b2b/account-history.schema.json [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+          "title": "Account Event Key",
+          "description": "Unique composite key identifier for the Account Event.",
+          "$ref": "https://ns.adobe.com/xdm/datatypes/b2b-source",
+          "meta:titleId": "account-history##xdm:accountKey##title##76911",
+          "meta:descriptionId": "account-history##xdm:accountKey##description##47311"
+        },
+        "xdm:accountKey": {
```
Comment (gauravsummer) [inline components/classes/b2b/account-history.schema.json:23]: Wouldn't we add "xdm:accountEventType" here as well?

File: components/fieldgroups/account/events/account-data-value-change.schema.json [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+    "of the License at https://creativecommons.org/licenses/by/4.0/"
+  ],
+  "$id": "https://ns.adobe.com/xdm/account/events/data-value-change",
+  "$schema": "http://json-schema.org/draft-06/schema#",
+  "title": "Data Value Change",
+  "type": "object",
+  "meta:tags": {
+    "b2bSchema": true
+  },
+  "meta:conditionalField": "xdm:eventType",
```
Comment (gauravsummer) [inline components/fieldgroups/account/events/account-data-value-change.schema.json]: What does this "xdm:eventType" signify here? Should it be  "xdm:accountEventType"  instead?

File: components/fieldgroups/account/events/account-data-value-change.schema.json [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+            "xdm:source": {
+              "title": "Source",
+              "type": "string",
+              "meta:titleId": "data-value-change##xdm:source##title##73041",
+              "meta:descriptionId": "data-value-change##xdm:source##description##10571"
+            },
+            "xdm:oldValueString": {
+              "title": "Old Value String",
+              "type": "string",
+              "description": "Old value.",
```
Comment (gauravsummer) [inline components/fieldgroups/account/events/account-data-value-change.schema.json]: We should include the description that this is the old value for String fields, number fields, datetime fields etc. Applies to all places following this.

File: components/fieldgroups/account/events/account-interesting-moment.schema.json [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+    "of the License at https://creativecommons.org/licenses/by/4.0/"
+  ],
+  "$id": "https://ns.adobe.com/xdm/account/events/account-interesting-moment",
+  "$schema": "http://json-schema.org/draft-06/schema#",
+  "title": "Account Interesting Moment",
+  "type": "object",
+  "meta:tags": {
+    "b2bSchema": true
+  },
+  "meta:conditionalField": "xdm:eventType",
```
Comment (gauravsummer) [inline components/fieldgroups/account/events/account-interesting-moment.schema.json]: What does this "xdm:eventType" signify here? Should it be "xdm:accountEventType" instead?

File: components/fieldgroups/account/events/account-score-change.schema.json [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+    "of the License at https://creativecommons.org/licenses/by/4.0/"
+  ],
+  "$id": "https://ns.adobe.com/xdm/account/events/account-score-change",
+  "$schema": "http://json-schema.org/draft-06/schema#",
+  "title": "Score Change",
+  "type": "object",
+  "meta:tags": {
+    "b2bSchema": true
+  },
+  "meta:conditionalField": "xdm:eventType",
```
Comment (gauravsummer) [inline components/fieldgroups/account/events/account-score-change.schema.json]: Same as above. Applies to all.
What does this "xdm:eventType" signify here? Should it be "xdm:accountEventType" instead?

**Review by gauravsummer — CHANGES_REQUESTED** [review]
Added review comments.

File: components/classes/b2b/account-history.schema.json (hunk 2 above)
Comment (chrispanella) [reply components/classes/b2b/account-history.schema.json]: added the ref and time series related metadata, removed the others.

File: components/classes/b2b/account-history.schema.json (hunk 3 above)
Comment (chrispanella) [reply components/classes/b2b/account-history.schema.json]: Removed the field

File: components/classes/b2b/account-history.schema.json (hunk 4 above)
Comment (chrispanella) [reply components/classes/b2b/account-history.schema.json:23]: I added the accountEventType here as well

File: components/fieldgroups/account/events/account-data-value-change.schema.json (hunk 5 above)
Comment (chrispanella) [reply components/fieldgroups/account/events/account-data-value-change.schema.json]: it should be now, I adjusted this on all areas

File: components/fieldgroups/account/events/account-data-value-change.schema.json (hunk 6 above)
Comment (chrispanella) [reply components/fieldgroups/account/events/account-data-value-change.schema.json]: Adjusted the descriptions to say string, number or date

File: components/fieldgroups/account/events/account-interesting-moment.schema.json (hunk 7 above)
Comment (chrispanella) [reply components/fieldgroups/account/events/account-interesting-moment.schema.json]: I adjusted this on all areas

File: components/fieldgroups/account/events/account-score-change.schema.json (hunk 8 above)
Comment (chrispanella) [reply components/fieldgroups/account/events/account-score-change.schema.json]: I adjusted this on all areas

**Pull request merged**